### ISSUE TITLE
Tor onion service is offline or out-of-date

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#1352]
We have a Tor hidden service mirror, thanks to @Nukama. However, it has fallen woefully behind the updates to the canonical website. (As of this writing, most pages on it are inaccessible.) Is there any way to automate the task of keeping it up-to-date? If not, perhaps we shouldn't list it in the footer of the site (at least not without some sort of disclaimer).

For reference, here's the URI: `http://qubesosmamapaxpa.onion/`

**Comment by marmarek** [comment]
@Nukama, I can setup some hook to call when the repository is updated.
On your side this would require exposing some address (either clearnet
or tor service), which will call _utils/update-submodules (or similar). The
script would take care of git tag verification. That exposed hook would
not need to parse any parameters, just fetch updates from github repo. 

-- 
Best Regards,
Marek Marczykowski-Górecki
Invisible Things Lab
A: Because it messes up the order in which people normally read text.
Q: Why is top-posting such a bad thing?

**Comment by axon-qubes** [comment]
We probably shouldn't wait for a reply from @Nukama on this.

**Comment by qubenix** [comment]
Maybe host the hidden service on the same server as the clearnet site? This will dramatically simplify this process and make an easier workflow for the dev team (if @Nukama will give the private key of course).

**Comment by marmarek** [comment]
Currently clearnet site is hosted on github pages, so not exactly possible to host hidden service from there. But cloning github pages (jekyll) is very simple.

**Comment by fortasse** [comment]
I can probably host this along side Whonix's hidden services. However, if you want to keep the same .onion domain, you'll have to get me the private key somehow. Otherwise, we can have another subdomain on kkkkkkkkkk63ava6.onion.

If it's just a matter of pulling a git repo every so often, that should be pretty easy to do.

**Comment by marmarek** [comment]
> I can probably host this along side Whonix's hidden services. However, if you want to keep the same .onion domain, you'll have to get me the private key somehow. Otherwise, we can have another subdomain on kkkkkkkkkk63ava6.onion.

Unfortunately, we don't have private key for the old address :/
@andrewdavidwong do you have any preference as for the address - from the list I've sent you?

> If it's just a matter of pulling a git repo every so often, that should be pretty easy to do.

I can also setup github hook for the repository - it will sent a simple POST request to configured URL. Possible problem here: it must be clearnet URL, onion addresses are not supported. If really needed, I could proxy it myself, but not sure if worth the effort.

**Comment by fortasse** [comment]
Our hidden service and clearnet sites are hosted on the same machine, so having the hook tie back to a clearnet address should be perfectly fine. I don't have too much experience with webhooks, however.

**Comment by marmarek** [comment]
It's easy: https://github.com/marmarek/signature-checker/blob/master/github-webhook - expose as CGI.
If unique script could be used for this particular purpose - probably you don't need to parse POST data at all. Some documentation: https://developer.github.com/v3/repos/hooks/#receiving-webhooks
And actual example event for git push: https://developer.github.com/v3/activity/events/types/#pushevent

**Comment by andrewdavidwong** [comment]
> Unfortunately, we don't have private key for the old address :/
@andrewdavidwong do you have any preference as for the address - from the list I've sent you?

No preference. There weren't any outstanding ones. Here were the slightly recognizable addresses:

```
qubesoseggc4nxja.onion
qubesos4rrrrz6n4.onion
qubesoseaaeexn7o.onion
qubesosxuuxxnkzb.onion
qubesosangieqolx.onion
qubesosnojpylaqe.onion
qubesosxydie7non.onion
qubesosgtgxgxdot.onion
qubesosprrwr4z3h.onion
qubesoscirect6nz.onion
```

IMO, using a subdomain on `kkkkkkkkkk63ava6.onion` would also be fine.

**Comment by mfc** [comment]
yep `qubesos4rrrrz6n4.onion` sounds good.

**Comment by fortasse** [comment]
Cool. If you want to email me the private key, my contact info is here: https://www.whonix.org/wiki/contact#For_Website_Issues

**Comment by fortasse** [comment]
I have qubesos4rrrrz6n4.onion up and running. @marmarek: I sent you an email to get the webhook / automation side part set up as well.

**Comment by marmarek** [comment]
Ok, webhook enabled. @andrewdavidwong do you have some pending edits to test if automation works?

**Comment by fortasse** [comment]
For full disclosure, the reason I emailed the webhook link directly is to prevent it from getting crawled by search engines and regenerating the site unnecessarily. If you email me directly and provide a half-decent reason for needing the endpoint, I'll happily provide it.

**Comment by fortasse** [comment]
If there are no editorial changes to make, adding a comment to robots.txt would likely work.

**Comment by marmarek** [comment]
If you want to prevent spurious calls, github provide a mechanism for verifying if request really come from github - `X-Hub-Signature` header: https://developer.github.com/v3/repos/hooks/#webhook-headers
But just for anti-crawlers, you can just check some simple property of received data - like if that is a POST request (very rare for crawlers), if `X-GitHub-Event` header is present etc.

Anyway, just keeping URL non-public is also some solution.

**Comment by fortasse** [comment]
Aah, good things to point out. Once we make sure everything's working I may go back and improve the actual webhook receiver, it's a bit inelegant at the moment.

**Comment by marmarek** [comment]
> If there are no editorial changes to make, adding a comment to robots.txt would likely work.

I'd like to check also submodules update. Just pushed some minor edit.

**Comment by fortasse** [comment]
Doesn't look like it worked. @marmarek, would you mind hopping in #whonix on OFTC to help me get this debugged?

**Comment by fortasse** [comment]
I've redone the webhook to essentially use a modified version of the post-receive hook in the [readme here](https://github.com/QubesOS/qubesos.github.io#notes). That should deal with the submodules properly.

**Comment by fortasse** [comment]
@marmarek joined me in IRC, and I believe everything is working now. I will make a note to clean up my implementation and potentially add it to _utils in the main repo.

**Comment by andrewdavidwong** [comment]
Added a footer link to the new hidden service. Thanks, @fortasse!

**Comment by anonmos1** [comment]
Can we have https://www.qubes-os.org/qubes-issues/ working on the onion too?

**Comment by andrewdavidwong** [comment]
> Can we have https://www.qubes-os.org/qubes-issues/ working on the onion too?

That one's hosted from a [different repo](https://github.com/QubesOS/qubes-issues/tree/gh-pages). The original intention was for it not to share the qubes-os.org domain at all, but because of the way the website is set up, there's an automatic redirect from https://qubesos.github.io/qubes-issues/ to https://www.qubes-os.org/qubes-issues/. Not sure how much trouble it'd be.

CC @fortasse @marmarek

**Comment by fortasse** [comment]
If we're cool with it just living at `qubesos4rrrrz6n4.onion/qubes-issues/` it shouldn't be too much trouble. We can use the existing webhook, I can just update the script to update both repos / folders.

**Comment by andrewdavidwong** [comment]
@fortasse: Sounds good to me. Thanks!

**Comment by fortasse** [comment]
@andrewdavidwong: http://qubesos4rrrrz6n4.onion/qubes-issues/ is up. Do you want to confirm the webhook works? It's exactly the same as the old webhook, just with a few more steps to pull and build https://github.com/QubesOS/qubes-issues/tree/gh-pages. I assume it will work fine, but more testing is never a bad thing. :)

**Comment by andrewdavidwong** [comment]
@fortasse Ok, I've pushed several commits to the `gh-pages` branch. They're not reflected on the onion version yet, but it might still be too soon.

**Comment by fortasse** [comment]
This might be a stupid question, but did you configure this repo to hit the same webhook as the main website repo?

**Comment by andrewdavidwong** [comment]
No, I've never done that before. (I think @marmarek did it for the main website repo.) How would I do that? Feel free to point me to documentation.

**Comment by fortasse** [comment]
https://developer.github.com/webhooks/creating/

TLDR: To set up a repository webhook on GitHub, head over to the Settings page of your repository, and click on Webhooks & services.

You should clone whatever settings are used for the [QubesOS/qubes-doc](https://github.com/QubesOS/qubes-doc) repo, I'm not sure what the exact settings are on the github-side.

**Comment by andrewdavidwong** [comment]
Oh, I don't have admin access to either repo, so I can't even view the Settings page. I think @marmarek will have to do it.

**Comment by fortasse** [comment]
Alright. I just hit the webhook manually, can you confirm the changes are reflected on qubesos4rrrrz6n4.onion/qubes-issues/ as expected? If hitting it manually works, it should have no issues once the webhook config gets updated on Github.

**Comment by fortasse** [comment]
I'm 95% sure it should work when the repo config is updated. Let me know if it still doesn't work once the webhook is configured on the github-side.

**Comment by andrewdavidwong** [comment]
Yep, looks good. Thanks, @fortasse!

**Comment by marmarek** [comment]
Webhook added. But there is some problem - it failed with "Service timeout", and also the last one for main repo failed at 2017-04-27 17:47:51 (the last one).

**Comment by fortasse** [comment]
Marek Marczykowski-Górecki:
> Webhook added. But there is some problem - it failed with "Service timeout", and also the last one for main repo failed at 2017-04-27 17:47:51 (the last one).
> 

That might be an issue on the github side, and it happens occasionally?
I see 200s for the webhook from the github hookshot in the logs, so I
think it works? I do see the HTTP 499 at 28/Apr/2017:09:59:50 +0200, but
everything after that is 200s.

I believe github's behavior is to retry webhooks that fail a given
number of times.

**Comment by anonmos1** [comment]
onion doesn't have the latest announcement

**Comment by fortasse** [comment]
anonmos1:
> onion doesn't have the latest announcement
> 

I don't see any POSTs to the webhook from Github. Did someone change the
config? The last log line is from 03/May/2017:15:10:01 +0200.

I'll update it manually in the meantime.

**Comment by marmarek** [comment]
According to github - this is also when the last POST was sent (successfully). And it is _after_ adding an announcement. So, something is wrong with the update script...

**Comment by fortasse** [comment]
Shoot, you're right - I had a typo in one of the directories. It should
be fixed now. My bad, thanks for pointing it out @anonmos1!

**Comment by anonmos1** [comment]
Unfortunately, the onion is not up to date

**Comment by fortasse** [comment]
Just updated it manually, and made a couple tweaks to the update
process. Hopefully that will prevent this from happening again.

Sorry for the trouble, and thanks for letting me know.

(Also, it looks like reply-by-email is broken on Github. If this double posts, sorry.)

**Comment by andrewdavidwong** [comment]
Thanks for taking care of it, @fortasse!

**Comment by anonmos1** [comment]
I still cannot see the latest announcements

**Comment by fortasse** [comment]
Hm, apparently the --incremental flag for Jekyll doesn't work like I thought. I've removed it, and confirmed the updates are now there. It seems like I was pulling the files from Git properly, but not generating the site completely...

Can you confirm @anonmos1?

**Comment by anonmos1** [comment]
Yes, all seems to be good

**Comment by anonmos1** [comment]
@fortasse

**Comment by fortasse** [comment]
Jekyll decided to be picky. I've got it sorted now, thanks for bringing it to my attention @anonmos1, and sorry for the delay.

**Comment by andrewdavidwong** [comment]
FYI @fortasse:

[On 2018-02-11 18:57, Gaijin wrote:](https://groups.google.com/d/msgid/qubes-users/aa4ee1ecef28508f855a29295ab7cd64%40riseup.net)
> The Qubes OS v3 and v2 onion services sites don't appear to have been
> updated since January when this announcement was made. Shouldn't they
> show the same content as the normal Qubes site? 
> 
> For example I don't see the news of the 4.0-rc4 release, or recent
> changes to the Docs.
>

**Comment by andrewdavidwong** [comment]
Bump/poke @fortasse:

[On 2018-03-12 02:26, Gaijin wrote:](https://groups.google.com/d/msgid/qubes-users/c56b5c1d344eeb12b4a69fc3b0975145%40riseup.net)
> On 2018-02-12 01:47, Andrew David Wong wrote:
>> On 2018-02-11 18:57, Gaijin wrote:
>>
>>> The Qubes OS v3 and v2 onion services sites don't appear to have been
>>> updated since January when this announcement was made. Shouldn't they
>>> show the same content as the normal Qubes site?
>>>
>>> For example I don't see the news of the 4.0-rc4 release, or recent
>>> changes to the Docs.
>>>
>>
>> Thanks for the report! Tracking:
>>
>> https://github.com/QubesOS/qubes-issues/issues/1352#issuecomment-364812018
>>
> 
> FYI: The Onion sites are still not syncing with the Qubes site. They're
> still stuck in January 2018...
> 

CC: @adrelanos

**Comment by fortasse** [comment]
Shoot, thanks again for the heads up - I've adjusted the script to address cases where the gems need upgrading, and to email me when it doesn't exit cleanly. Thanks for your patience!

**Comment by andrewdavidwong** [comment]
Thanks!

**Comment by marmarek** [comment]
Looks like tor onion service is down.

**Comment by adrelanos** [comment]
This is unlikely being fixed anytime soon.

https://forums.whonix.org/t/whonix-org-webmaster-fortasse-stepped-down

**Comment by andrewdavidwong** [comment]
That's very sad news. Thank you for letting us know.

In the meantime, I suggest that users who want to update over Tor change their repo definitions to use the non-onion domains (i.e., `yum.qubes-os.org` and `deb.qubes-os.org`).

**Comment by mig5** [comment]
@andrewdavidwong @marmarek thought it worth mentioning that I'm now the sysadmin of Whonix infrastructure (and ergo, the Qubes OS onions). Please verify with Patrick if you need, or see this [forum thread](https://github.com/QubesOS/qubes-issues/issues/1352)

If you encounter issues with the Qubes onions please do let me know - my GPG key is at https://keybase.io/mig5 and I can be reached via email, Whonix forums/phabricator, Keybase chat, Signal (num on request), or ping me in Github.

We also have a staging environment now for testing any potential changes to our (Tor, and other) configs, as maintaining stability is a priority for me.

**Comment by marmarek** [comment]
> this forum thread

You have linked _this_ issue here...

Anyway, good to hear that, thanks!

**Comment by mig5** [comment]
Classic ctrl+shift+v fail between AppVMs :)

Link is at https://forums.whonix.org/t/new-sysadmin-saying-hello/5446

Thanks!

**Comment by andrewdavidwong** [comment]
Welcome, @mig5! Thanks for letting us know.

**Comment by andrewdavidwong** [comment]
@mig5, a user has reported (and I have confirmed) that both of the Qubes website onion links are not working (connection times out):

http://qubesos4rrrrz6n4.onion/
http://sik5nlgfc5qylnnsr57qrbm64zbdx6t4lreyhpon3ychmxmiem7tioad.onion/

**Comment by adrelanos** [comment]
I restarted the Tor daemon. Let's hope that helps. That's of course just a workaround which helps temporarily so it could go down later again. Not a fix.

Depending on the outcome of https://forums.whonix.org/t/whonix-project-looking-for-web-server-sponsor/6971 it might soon go down and stay down for a long time or forever.

**Comment by andrewdavidwong** [comment]
@unman, the new Qubes website Tor onion service is not available for me. Is this the correct URL?

http://www.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/

(Other onion URLs are working for me.)

**Comment by andrewdavidwong** [comment]
> @unman, the new Qubes website Tor onion service is not available for me. Is this the correct URL?
> 
> http://www.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/
> 
> (Other onion URLs are working for me.)

It's back! Looks like it was just temporary, and the website is up-to-date, so I'll close this issue. Thanks again, @unman.

**Comment by NipponBill** [comment]
The onion site is out of sync again.

For example, the News page's last update was November 2019: http://qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/news/

**Comment by andrewdavidwong** [comment]
> The onion site is out of sync again.
> 
> For example, the News page's last update was November 2019: http://qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/news/

FYI @mig5 @adrelanos

**Comment by adrelanos** [comment]
Not hosted by me.

**Comment by andrewdavidwong** [comment]
> Not hosted by me.

Oops, was I supposed to mention @unman instead?

**Comment by NipponBill** [comment]
The .onion site updated to news articles through January 2...which is a bit better, but it's still not in sync with the regular website...at least for News. @unman

**Comment by unman** [comment]
On Wed, Jan 22, 2020 at 01:15:31AM -0800, NipponBill wrote:
> The .onion site updated to news articles through January 2...which is a bit better, but it's still not in sync with the regular website...at least for News. @unman 
> 
You're right, and I dont know why not.
I'll manually update it just now.

**Comment by NipponBill** [comment]
This is still an issue. I notice it more now that the Tor browser prompts the user to "Always Prioritize Onions", and if the user opts in, they get transferred to the .onion site.

Ex. [Today's announcement](https://www.qubes-os.org/news/2020/08/20/new-community-forum-for-qubes-os-users/) of the Discord forum. This turns into a 404 Not Found page. 
@unman

**Comment by andrewdavidwong** [comment]
@NipponBill, is this also affecting the repos or only the website?

**Comment by NipponBill** [comment]
@andrewdavidwong I have noticed this with the News and Docs on the **website** as I monitor https://www.qubes-os.org/ for updates. I have not tracked the state of the repos.

**Comment by Solomon1732** [comment]
Currently the repos and website hidden services fail to connect. The Tor browser at least returns code 0xF0 when I attempt to connect to any of them. I found them to be down for at least a few days now. I don't have this issue with other hidden services. It seems specific to Qubes services. Are they down, perhaps?

I get them by going to https://www.qubes-os.org/ . Then I let the browser automatically redirect me to the following:
http://qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/
http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/
http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/

Copying the URL from the Qubes repo for the website doesn't change the result.
https://github.com/QubesOS/qubesos.github.io/blob/1b30f7e0d906ff7377341c5dab8f7e13f3086d05/_includes/head.html#L20

**Comment by unman** [comment]
On Mon, Nov 22, 2021 at 10:21:15AM -0800, Solomon1732 wrote:
> Currently the repos and website hidden services fail to connect. The Tor browser at least returns code 0xF0 when I attempt to connect to any of them. I found them to be down for at least a few days now. I don't have this issue with other hidden services. It seems specific to Qubes services. Are they down, perhaps?
> 
> I get them by going to https://www.qubes-os.org/ . Then I let the browser automatically redirect me to the following:
> http://qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/
> http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/
> http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/
> 
> Copying the URL from the Qubes repo for the website doesn't change the result.
> https://github.com/QubesOS/qubesos.github.io/blob/1b30f7e0d906ff7377341c5dab8f7e13f3086d05/_includes/head.html#L20
> 

Strange, because there was a thread on the Forum saying the same, but I
couldn't reproduce that. Nor have I noticed this, certainly not for "at least a
few days": that is, whenever I have used the onion sites, they have been
available.

**Comment by andrewdavidwong** [comment]
Received another report:

> Onionsite Not Found
> 
> The most likely cause is that the onionsite is offline. Contact the onionsite administrator.
> 
> Details: 0xF0 — The requested onion service descriptor can't be found on the hashring and therefore the service is not reachable by the client.
> 
> 
> hello, looks like qubes os site .onion is broken, last 2 days, fyi

**Comment by Solomon1732** [comment]
When I try to reach the hidden service I get this:
> Onionsite Not Found
>
> The most likely cause is that the onionsite is offline. Contact the onionsite administrator.
>
> Details: 0xF0 — The requested onion service descriptor can't be found on the hashring and therefore the service is not reachable by the client.
I tried every few hours. No luck.

**Comment by unman** [comment]
On Tue, Jul 12, 2022 at 03:48:32PM -0700, Solomon1732 wrote:
> When I try to reach the hidden service I get this:
> > Onionsite Not Found
> >
> > The most likely cause is that the onionsite is offline. Contact the onionsite administrator.
> >
> > Details: 0xF0 ??? The requested onion service descriptor can't be found on the hashring and therefore the service is not reachable by the client.
> I tried every few hours. No luck.
> 

I'm advised there is currently a hardware issue.

**Comment by apparatius** [comment]
The rpm files for testing templates that are present in clearnet Qubes OS repository are missing in onion repository:
https://yum.qubes-os.org/r4.2/templates-itl-testing/
https://yum.qubes-os.org/r4.2/templates-community-testing/
http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.2/templates-itl-testing/
http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.2/templates-community-testing/

**Comment by apparatius** [comment]
It seems that the onion sync was triggered on 7 Jan and some of templates-itl-testing templates were uploaded but then it stopped.
And after this the new package updates are not uploaded as well, e.g. `kernel-latest-6.6.9-1.qubes.fc37.x86_64.rpm` form current-testing is not uploaded to the onion repo:
https://yum.qubes-os.org/r4.2/current-testing/dom0/fc37/rpm/kernel-latest-6.6.9-1.qubes.fc37.x86_64.rpm
Is missing here:
http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.2/current-testing/dom0/fc37/rpm/

Maybe there in not enough free space to sync or something else.

**Comment by NipponBill** [comment]
The onion site is out of sync again. 

I was surprised to find that the Fedora 39 templates had been made available 12 days ago because I'd been monitoring the .onion site's News page.

**Comment by Solomon1732** [comment]
The onion site is down. This is what I get when I test it with the Tor browser:
1. I go to https://www.qubes-os.org/
2. I use Tor browser's redirect to onion button (the purple ".onion available) one in the address bar).
3. I get this:
># Onion site not found
>The most likely cause is that the onion site is offline. Contact the onion site administrator.

**Comment by unman** [comment]
Apparently down for maintenance.

**Comment by Solomon1732** [comment]
The onion site is down again. It blocks me from updating my templates and standalones since I update them through it and over tor. Is it down for maintenance again? I've been waiting a few days before asking this.

**Comment by unman** [comment]
Yes. I'm expecting it to come up on a new server, with redundancy, soon.

**Comment by Solomon1732** [comment]
The onion site is down again and has been so for the last few days. Same story as hereinabove https://github.com/QubesOS/qubes-issues/issues/1352#issuecomment-2465555054

**Comment by NipponBill** [comment]
@unman The News page on the Onion site hasn't updated since July 1. It is quite out of date now.

**Comment by unman** [comment]
I froze any updates until the RST conversion is rolled out.

**Comment by andrewdavidwong** [comment]
> I froze any updates until the RST conversion is rolled out.

News posts are not part of the RST documentation conversion, but if it was easier just to freeze all updates, then that's understandable. It looks like the migration should be complete soon.

**Comment by faklapant23** [comment]
@unman some files in the onion repository are missing.
I can't update the dom0 in Qubes OS 4.2 because of the missing package:
http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.2/current-testing/host/fc37/rpm/linux-firmware-whence-20251011-1.fc37.noarch.rpm
And there is no linux-firmware-whence-* packages at all that are available for dom0 in the onion repo, but they exist in the clearnet repo e.g.:
https://yum.qubes-os.org/r4.2/current-testing/host/fc37/rpm/linux-firmware-whence-20251011-1.fc37.noarch.rpm

**Comment by unman** [comment]
Bad mirroring, now fixed. Thanks for pointing this out.

**Comment by faklapant23** [comment]
@unman some templates are missing in the onion repository.
I guess this is done to save the space since older templates are rarely needed, but some templates are missing entirely, for example, archlinux, kali, gentoo and centos for Qubes OS 4.2 are missing in the templates-community and templates-community-testing repos.

**Comment by NipponBill** [comment]
@unman The **News** page on the Onion site hasn't updated since at least Sept. 30, but I see that the Documentation is still an issue.

**Comment by Euwiiwueir** [comment]
FYI I believe http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion is offline; first noticed yesterday.

edit: Per https://forum.qubes-os.org/t/qubesos-onion-repo-unreachable/38604/4 the onion server is back online :+1:

**Comment by Solomon1732** [comment]
I think http://yum.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/ is offline again. I thought it was my internet or the Tor network, since I can't update qubes through onion domains. Then I tried surfing it directly through the Tor browser. It writes:
> # Onion site not found
> The most likely cause is that the onion site is offline. Contact the onion site administrator.

**Comment by unman** [comment]
Works for me

**Comment by Solomon1732** [comment]
I checked again and it's working for me again.

By the way, I waited some hours before my previous message, so I don't know why the temporary hiccup.

**Comment by Solomon1732** [comment]
The onion service is offline again. I get this:
> # Onion site not found
> The most likely cause is that the onion site is offline. Contact the onion site administrator.

I can't update my system since I use onion domains for the qubes repos.

**Comment by rustybird** [comment]
https://forum.qubes-os.org/t/qubes-r4-2-repo-missing-kernel-6-18-26/41163
